File: December - 08/cafeteria_queue_challenge.R
input <- scan("stdin", quiet = TRUE)
n <- input[1]
students <- input[2:(1 + n)]
sandwiches <- input[(2 + n):(1 + 2 * n)]

count0 <- sum(students == 0)
count1 <- n - count0

for (i in 1:n) {
  if (sandwiches[i] == 0) {
    if (count0 > 0) {
      count0 <- count0 - 1
    } else {
      break
    }
  } else {
    if (count1 > 0) {
      count1 <- count1 - 1
    } else {
      break
    }
  }
}

cat(count0 + count1)

File: December - 08/CafeteriaQueueChallenge.java
public class CafeteriaQueueChallenge {

    public static void main(String[] args) {
        int[] students = {1, 1, 0, 0};
        int[] sandwiches = {0, 1, 0, 1};
        int count0 = 0, count1 = 0;

        for (int s : students) {
            if (s == 0) count0++;
            else count1++;
        }

        for (int sw : sandwiches) {
            if (sw == 0) {
                if (count0 > 0) count0--;
                else break;
            } else {
                if (count1 > 0) count1--;
                else break;
            }
        }
        int unableToEat = count0 + count1;
        System.out.println(unableToEat);
    }
}
